src/chainrules.jl

```diff
@@ -31,49 +36,49 @@ ChainRulesCore.@scalar_rule(trigamma(x), polygamma(2, x))
 ChainRulesCore.@scalar_rule(
     besselj(ν, x),
     (
-        ChainRulesCore.@thunk(error("not implemented")),
+        ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO),
         (besselj(ν - 1, x) - besselj(ν + 1, x)) / 2
     ),
 )
 ChainRulesCore.@scalar_rule(
     besseli(ν, x),
     (
-        ChainRulesCore.@thunk(error("not implemented")),
+        ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO),
         (besseli(ν - 1, x) + besseli(ν + 1, x)) / 2,
     ),
 )
 ChainRulesCore.@scalar_rule(
     bessely(ν, x),
     (
-        ChainRulesCore.@thunk(error("not implemented")),
+        ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO),
         (bessely(ν - 1, x) - bessely(ν + 1, x)) / 2,
     ),
 )
 ChainRulesCore.@scalar_rule(
     besselk(ν, x),
     (
-        ChainRulesCore.@thunk(error("not implemented")),
+        ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO),
         -(besselk(ν - 1, x) + besselk(ν + 1, x)) / 2,
     ),
 )
 ChainRulesCore.@scalar_rule(
     hankelh1(ν, x),
     (
-        ChainRulesCore.@thunk(error("not implemented")),
+        ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO),
         (hankelh1(ν - 1, x) - hankelh1(ν + 1, x)) / 2,
     ),
 )
 ChainRulesCore.@scalar_rule(
     hankelh2(ν, x),
     (
-        ChainRulesCore.@thunk(error("not implemented")),
+        ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO),
         (hankelh2(ν - 1, x) - hankelh2(ν + 1, x)) / 2,
     ),
 )
 ChainRulesCore.@scalar_rule(
     polygamma(m, x),
     (
-        ChainRulesCore.@thunk(error("not implemented")),
+        ChainRulesCore.DoesNotExist(),
```

Comment:
Why wasn't it already the case?

Reply:
I guess I did not pay enough attention when I moved the rules from ChainRules and just replaced all `NaN` with `@thunk(error("not implemented"))` (https://github.com/JuliaDiff/ChainRules.jl/blob/bf54b2f0e77fb05d3bc42004c448b3db0432f9cd/src/rulesets/packages/SpecialFunctions.jl#L57).